### DISCUSSION TITLE
Render Drawer in cusotm container

### DISCUSSION THREAD

**Discussion by isBatak** [chakra-ui/chakra-ui#8886] [Ideas]
### Description

Is it possible ot add OTB support for Drawer beeing rendered inside custom container?

Example:
https://ant.design/components/drawer#components-drawer-demo-render-in-current

Current hacky solution:
https://codesandbox.io/s/chakra-ui-drawer-as-sidebar-ndhzno
Only `placement="left"` is working correctly, other positions are broken. 

### Problem Statement/Justification

This is useful feature to have considering other libraries are supporting it.

### Proposed Solution or API

When `containerRef` is provided Drawer should apply different styles 
```
<Drawer
  portalProps={{
    containerRef: imageWrapperRef
  }}
>
```

**Comment by Chukwuamaka** [comment]
I agree with @isBatak about the feature being necessary. My use case is that I need to edit content and preview those changes on the side in a container that mimics a phone. Within that container, I'm expected to display some information in a bottom drawer (a.k.a. bottom sheet) but I can't achieve that since the drawer takes up the whole screen.

I'm going to try implementing this manually until the Drawer component supports this feature.
